entity-framework/core/providers/cosmos/limitations.md

```diff
@@ -2,29 +2,27 @@
+The Azure Cosmos DB database provider targets the Azure Cosmos DB NoSQL store, which is a [document database](https://en.wikipedia.org/wiki/Document-oriented_database). Most EF Core providers target [relational databases](https://en.wikipedia.org/wiki/Relational_database). Document databases and relational databases behave in fundamentally different ways. EF Core does not attempt to hide these differences; rather EF Core provides common patterns that can be used successfully across both kinds of database, together with features tailed to a particular provider that follow best practices for a given type of database. If a feature of EF Core is a pit-of-failure for a certain kind of database, then typically the database provider will not implement that feature, and instead help steer uses towards a [pit-of-success](/archive/blogs/brada/the-pit-of-success) approach.
 
-These are some of the commonly requested features:
+Common EF Core patterns that either do not apply, or are a pit-of-failure, when using a document database include:
 
-- [`Include` support](https://github.com/dotnet/efcore/issues/16920)
-- [`Join` support](https://github.com/dotnet/efcore/issues/17314)
-- [Collections of primitive types support](https://github.com/dotnet/efcore/issues/14762)
+- Schema migration is not supported, since there is no defined schema for the documents. However, there could be other mechanisms for dealing with evolving data shapes that do make sense with Azure Cosmos DB NoSQL, For example, [Schema versioning pattern with Cosmos DB](https://github.com/dotnet/efcore/issues/23753), and [Cosmos data migration](https://github.com/dotnet/efcore/issues/11099).
+- Reverse-engineering (scaffolding) a model from an existing database is not supported. Again, this is not supported because there is no defined database schema to scaffold from. However, see [Use shape of documents in the Cosmos database to scaffold a schema](https://github.com/dotnet/efcore/issues/30290).
+- Schema concepts defined on the EF model, like indexes and constraints, are ignored when using a document database, since there is no schema. Note that Azure Cosmos DB NoSQL performs [automatic indexing of documents](/azure/cosmos-db/index-overview).
+- Loading graphs of related entities from different documents is not supported. Document databases are not designed to perform joins across many documents; doing so would be very inefficient. Instead, it is more common to denormalize data so that everything needed is in one, or a small number, of documents. However, there are some forms of cross-document relationships that could be handled--see [Limited Include support for Cosmos](https://github.com/dotnet/efcore/issues/16920#issuecomment-989721078).
 
 ## Azure Cosmos DB SDK limitations
 
-- Only async methods are provided
+Some functionality supported by most EF Core database providers are not supported by the Azure Cosmos DB SDK. This includes synchronous APIs.
```

Comment:
Not sure why we have this separate category of limitations here - I don't think it's very relevant which limitations come from the EF provider (e.g. no migrations) and which come from the underlying Cosmos SDK (no async). Should we just merge them?

Reply:
@AndriySvyryd?

Reply:
It made sense previously, when the provider limitations were temporary. Now they can be merged